### PR TITLE
Bindu, Keerthi| MOBN-2155 | Add. new form Clinical Pharmacy Assessment

### PR DESCRIPTION
**Pull request by himabindu1238**

### CODE REVIEW

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+                "CPA, Comments on drug-drug interaction","CPA, Type of adverse drug reaction (ADR)","CPA, Comments on adverse drug reaction","CPA, Comments on contraindications",
+                "CPA, Comments on untreated condition","CPA, Comments on therapeutic duplication","CPA, Comments on drug without indication","CPA, Therapeutic drug monitoring for",
+                "CPA, Comments on therapeutic drug monitoring","CPA, Comments on alternative recommended medications from MSF standard list","CPA, Comments on laboratory investigation required",
+                "CPA, Intervention accepted by the prescriber?","CPA, Comments on the accepted intervention","CPA, Why intervention not accepted by the prescriber?")
+                and concept_name_type = "FULLY_SPECIFIED")
+                AND
+                cn.concept_id IN (select concept_id from concept_name where name = "CPA, Clinical pharmacy interventions");
+            </sqlCheck>
+        </preConditions>
+        <comment>Adding Set Members to Clinical Pharmacy Assessment</comment>
```
Comment (supriyatw) [inline openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml:386]: This is for clinical pharmacy interventions

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+            set @child2_concept_id = 0;
+            select concept_id into @child1_concept_id from concept_name where name ="Yes" and concept_name_type ="FULLY_SPECIFIED" and locale ="en" and voided = 0;
+            select concept_id into @child2_concept_id from concept_name where name ="No" and concept_name_type ="FULLY_SPECIFIED" and locale ="en" and voided = 0;
+            call add_concept_answer(@concept_id, @child1_concept_id, 1);
+            call add_concept_answer(@concept_id, @child2_concept_id, 2);
+        </sql>
+    </changeSet>
+
+
+    <changeSet id="AMMAN_CONFIG_20230810160135523" author="Bindu">
```
Comment (supriyatw) [inline openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml:262]: To have good readbility maintain the order of change sets according to the design

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml (hunk 1 above)
Comment (supriyatw) [inline openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml:386]: This is for clinical pharmacy interventions

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                select COUNT(*) from concept_set cs
+                join concept_name cn
+                on cs.concept_id = cn.concept_id
+                where cs.concept_id IN (select concept_id from concept_name where name ="CPA, Clinical Pharmacy Assessment"  AND concept_name_type = "FULLY_SPECIFIED")
+                and cs.concept_set IN (select concept_id from concept_name where name = "All Observation Templates");
+            </sqlCheck>
+        </preConditions>
+        <comment>Adding set members to all observation templates</comment>
```
Comment (supriyatw) [inline openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml:272]: To have good readbility maintain the order of change sets according to the design

**Review by supriyatw — CHANGES_REQUESTED** [review]

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml (hunk 1 above)
Comment (keerthi-magantii) [reply openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml]: done

File: openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml (hunk 3 above)
Comment (keerthi-magantii) [reply openmrs/migrations/clinicalForms/ClinicalPharmacyAssessment.xml]: done

**Review by supriyatw — APPROVED** [review]

**Pull request merged**